crates/router/src/core/payments/operations/payment_update.rs

```diff
@@ -218,6 +221,14 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
         )
         .await?;
 
+        let billing_details: Option<Address> = billing_address.as_ref().map(From::from);
+        payment_intent.billing_details = billing_details
+            .clone()
+            .async_and_then(|_| async {
+                create_encrypted_data(key_store, billing_details.clone()).await
+            })
+            .await;
+
```

Comment:
here we are updating payment_intent in get trackers, and in other places we are updating payment_intent in update trackers. This can be made consistent